### DISCUSSION TITLE
New Command `:popover` (Hover Editor Plugin) for output channel Open files

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#186] [Ideas - on hold]
I'm starting a new discussion for something that @FelipeRearden wrote here: https://github.com/Taitava/obsidian-shellcommands/issues/143#issuecomment-1067879175
> # New Command `:popover` (Hover Editor Plugin)
> 
> Hello @Taitava !!!!
> 
> Just to let you know that we have a new plugin in beta that has a feature that I think will change the way that we use Obsidian forever.
> 
> https://github.com/nothingislost/obsidian-hover-editor
> 
> Basically, it will add a new type of way to open notes making them floating in the screen.
> 
> This way, we gonna have this options:
> 
>     * Open a file
> 
>     * Open a file in a new panel vertically
> 
>     * Open a file in a new panel horizontally
> 
>     * Open a file as popover
> 
> 
> This way, I think would be great to have a new command aligned to this new view :)
> 
> I know in very early to write about this ... but since I'm testing this new plugin (and creating workflows around it) I decide to write this now to not forget in the future :)
> 
> Take a look at the plugin, I think you will gonna like it !!!
> 
> Have a great day!!!!!

**Comment by Taitava** [comment]
I didn't try the plugin yet (maybe I'll wait for it to be included in the community plugins list), but the demo video looks cool! 😎 

I am interested in making the _Open a file_ (= will be renamed to _Open files_ in the future) output channel to support this new plugin. In practise, to be able to do this, _Hover editor_ would need to define a public API for its functionality, and @Quorafind seems to have opened [a public API feature request](https://github.com/nothingislost/obsidian-hover-editor/issues/36) a couple of hours ago. I'll add my comment there soon, too.

As for the `:popover` definition part, it's a good suggestion, but I think I'll use something that comes from the name of the plugin, i.e. `:hover` or `:hover-editor`. Not sure yet. The idea is that when it's a feature that depends on another plugin, it needs to be very transparent to the user what plugin they will need to install, and by using (part of) that plugin's name in the feature itself, it works as a good reminder of what plugin this feature relates to. 👍

---

> * Open a file in a new panel vertically
> 
> * Open a file in a new panel horizontally

Like we discussed earlier, [I haven't found a way in the public API of Obsidian to decide the splitting direction](https://github.com/Taitava/obsidian-shellcommands/issues/143#issuecomment-1046117915). So for now, choosing a vertical/horizontal split when opening a file in a new pane, won't unfortunately happen. 😞 (You can remind me sometime to write a feature request for it in Obsidian's forums 🙂).

---

Overall, really good idea from you @FelipeRearden, and it will work nicely together with the future development that will allow opening multiple files at the same time. (Hmm, it seems I haven't written about making it possible to open multiple files at once here in GitHub, but there's [a short mention in the documentation](https://publish.obsidian.md/shellcommands/Output+handling/Output+channel+-+Open+a+file#Restricted+support+for+line-breaks)).

**Comment by Quorafind** [comment]
https://github.com/nothingislost/obsidian-hover-editor/pull/41#issuecomment-1069409086 As reference

**Reply by Taitava** [reply]
Thank you! 🙂 I'll quote @nothingislost's message here:

> Thank you for the contribution but I'm not ready to commit to an API exposure style or interface at this stage. For now you can use the internal `this.app.plugins.plugins["obsidian-hover-editor"].spawnPopover` but I would not suggest building and publishing plugins around the usage of that as things are still quite a bit in flux.

This makes perfectly sense. As the plugin is really new, API changes can happen. I will wait until a real public API is published, like is suggested.

# I'm moving this discussion to _Ideas - on hold_

If a public API becomes available later, this discussion can be moved back to the active _Ideas_ category and planned further.
